storage/secondary.go

```diff
@@ -50,18 +64,88 @@ func (s *secondaryQuerier) LabelNames() ([]string, Warnings, error) {
 	return names, w, nil
 }
 
-type secondarySeriesSet struct {
-	genericSeriesSet
+func (s *secondaryQuerier) createFn(asyncSet genericSeriesSet) func() (genericSeriesSet, bool) {
+	s.asyncSets = append(s.asyncSets, asyncSet)
+	curr := len(s.asyncSets) - 1
+	return func() (genericSeriesSet, bool) {
+		s.once.Do(func() {
+			// At first create invocation we iterate over all sets and ensure its Next() returns some value without
+			// errors. This is to ensure we support consistent partial failures.
+			var errs tsdb_errors.MultiError
+			for i, set := range s.asyncSets {
+				if set.Next() {
```

Comment:
Hmm, might this lose us the first series?

Reply:
Yea, not as we lazySeriesSet cares for that